client/shared/src/main/scala/org/http4s/client/oauth1/SignatureAlgorithm.scala

```diff
@@ -70,30 +68,51 @@ trait SignatureAlgorithm {
     */
   def name: String
 
+  @deprecated("Use generateBase64[F[_]: MonadThrow] instead", "0.23.28")
```

Comment:
I'm considering this deprecation and can't fully agree with it. Since SignatureAlgorithm is primarily used as internal machinery in OAuth1, and users typically just select one from the existing algorithms, why should we deprecate it? Especially when we're not changing anything in the OAuth1 implementation.

Reply:
I think I got a bit carried away with the guidance from the contributing - https://http4s.org/contributing.html#safe-constructors

I am happy to remove those and keep the default behaviour. 

I had to add a ProblemFilters for the new method that was added to the trait. Since it's a public api, I was told that it's a breaking change since it doesn't have a default implementation. 

Would you happen to have any suggestions on how to tackle this?

Reply:
@danicheg having thought about it a bit more, would it make sense to create a new public trait for the AlgorithmSignature, which implements just the new methods. This way we can deprecate the existing trait and remove the StackOverflow issue?